client/src/model/apis/navigationApi/navigationApi.js

```diff
@@ -30,7 +30,7 @@ export const createNavigationApi = ({ store }) => {
     Boolean(matchPath(route, getPathName()));
 
   const setRoute = async (route, params = {}) => {
-    if (isRouteOpenedRightNow(route)) {
+    if (isRouteOpenedRightNow(route) && (route !== routes.openExercise)) {
```

Comment:
try to fix

Reply:
leave TODO (saying it's a hack etc.) above this `if` and let it be.
format like `TODO [github_username] [message]`